build-tools/packages/version-tools/src/test/internalVersionScheme.test.ts

```diff
@@ -125,65 +125,36 @@ describe("internalScheme", () => {
 			assert.isFalse(result);
 		});
 
-		it(">=2.0.0-internal.1.0.0 <2.0.0-internal.1.1.0 is internal", () => {
-			const input = `>=2.0.0-internal.1.0.0 <2.0.0-internal.1.1.0`;
-			assert.isTrue(isInternalVersionRange(input));
-		});
-
-		it(">=2.0.0-rc.1.0.0 <2.0.0-rc.1.1.0 is internal", () => {
-			const input = `>=2.0.0-rc.1.0.0 <2.0.0-rc.1.1.0`;
-			assert.isTrue(isInternalVersionRange(input));
-		});
-
-		// This test case should fail but it doesn't. "Fluid internal version ranges" should always have a prerelease
-		// identifier that matches between the upper and lower bound. It's skipped because I think the case it guards
-		// against isn't likely, so I don't think it's worth the cost of fixing it.
-		//
-		// The reason the code behaves wrong is because it only checks the lower bound of the range to see if it's an
-		// internal version. If the lower bound version is internal, then the function returns true.
-		it.skip(">=2.0.0-internal.1.0.0 <2.0.0-rc.1.1.0 is not internal", () => {
-			const input = `>=2.0.0-internal.1.0.0 <2.0.0-rc.1.1.0`;
-			assert.isFalse(isInternalVersionRange(input));
-		});
-
-		it(">=2.0.0-internal.2.2.1 <2.0.0-internal.3.0.0 is internal", () => {
-			const input = `>=2.0.0-internal.2.2.1 <2.0.0-internal.3.0.0`;
-			assert.isTrue(isInternalVersionRange(input));
-		});
-
-		it(">=2.0.0-alpha.2.2.1 <2.0.0-alpha.3.0.0 is not internal", () => {
-			const input = `>=2.0.0-alpha.2.2.1 <2.0.0-alpha.3.0.0`;
-			assert.isFalse(isInternalVersionRange(input));
-		});
-
-		it(">=2.0.0-alpha.2.2.1 <2.0.0-alpha.3.0.0 is internal when allowAnyPrereleaseId is true", () => {
-			const input = `>=2.0.0-alpha.2.2.1 <2.0.0-alpha.3.0.0`;
-			assert.isTrue(isInternalVersionRange(input, true));
-		});
-
-		it(">=2.0.0-dev.2.2.1.12345 <2.0.0-dev.3.0.0 is internal when allowAnyPrereleaseId is true", () => {
-			const input = `>=2.0.0-dev.2.2.1.12345 <2.0.0-dev.3.0.0`;
-			assert.isTrue(isInternalVersionRange(input, true));
-		});
-
-		it(">=1.0.0 <2.0.0 is not internal", () => {
-			const input = `>=1.0.0 <2.0.0`;
-			assert.isFalse(isInternalVersionRange(input));
-		});
-
-		it(">=2.0.0-2.2.1 <2.0.0-3.0.0 is not internal", () => {
-			const input = `>=2.0.0-2.2.1 <2.0.0-3.0.0`;
-			assert.isFalse(isInternalVersionRange(input));
-		});
-
-		it("^2.0.0-internal.2.2.1 is not internal", () => {
-			const input = `^2.0.0-internal.2.2.1`;
-			assert.isFalse(isInternalVersionRange(input));
-		});
-
-		it("~2.0.0-internal.2.2.1 is not internal", () => {
-			const input = `~2.0.0-internal.2.2.1`;
-			assert.isFalse(isInternalVersionRange(input));
+		const cases: [
+			isInternal: boolean,
+			input: string,
+			allowAnyPrereleaseId: boolean | undefined,
+		][] = [
+			[true, ">=2.0.0-internal.1.0.0 <2.0.0-internal.1.1.0", undefined],
+			[true, ">=2.0.0-rc.1.0.0 <2.0.0-rc.1.1.0", undefined],
+			[false, ">=2.0.0-internal.1.0.0 <2.0.0-rc.1.1.0", undefined],
```

Comment:
nit: FWIW, I find this format harder to read and understand. I think it's mostly that the test case "name" isn't easy to read. If I see "~2.0.0-internal.2.2.1 is not internal" I know what the test is checking without looking at the code.

Reply:
I think you are referring to the displayed test names shown in test output or test explorers. Such as:
```log
      ✔ >=2.0.0-alpha.2.2.1 <2.0.0-alpha.3.0.0 is internal when allowAnyPrereleaseId is true
      ✔ >=2.0.0-dev.2.2.1.12345 <2.0.0-dev.3.0.0 is internal when allowAnyPrereleaseId is true
      ✔ >=1.0.0 <2.0.0 is not internal
      ✔ >=2.0.0-2.2.1 <2.0.0-3.0.0 is not internal
      ✔ ^2.0.0-internal.2.2.1 is not internal
      ✔ ~2.0.0-internal.2.2.1 is not internal
```
Those are preserved with this change. The above is from my local test run. Please let me know if you meant that or something else.

Reply:
Yeah, saw that. I was referring to the source, though. It's probably just me - I can't look at that table and parse it easily.

Reply:
I agree for the source. It isn't just you. I like the table because it addresses some issues with typos and making the file too long to see all of the cases which makes it hard to see what isn't there. There are two alternate patterns.
1. Instead of array of arrays use array of records which has the parameters named and is a bit easier to follow a single line but adds verbosity to the table which makes it bigger again.
2. Use a helper function and just call it a bunch of times in a row.